packages/core/src/components/fields/SchemaField.js

```diff
@@ -71,17 +71,41 @@ function Label(props) {
   );
 }
 
-function LabelInput(props) {
-  const { id, label, onChange } = props;
-  return (
-    <input
-      className="form-control"
-      type="text"
-      id={id}
-      onBlur={event => onChange(event.target.value)}
-      defaultValue={label}
-    />
-  );
+class LabelInput extends React.Component {
+  constructor(props) {
+    super();
+    this.state = {
+      value: props.label,
+    };
+  }
+  render() {
+    const { id, label, onChange, schema, registry } = this.props;
+
+    if (!schema) {
+      return (
+        <input
+          className="form-control"
+          type="text"
+          id={id}
+          onBlur={event => onChange(event.target.value)}
+          defaultValue={label}
+        />
+      );
+    }
+
+    return (
+      <SchemaField
+        schema={schema}
+        formData={this.state.value}
+        registry={registry}
+        onChange={value => {
```

Comment:
Since we're always calling `onChange(value);` in the `onChange` prop, is there even a need to store `this.state.value` now? Could we just use something like `this.props.value` to get the current value of the field?

Reply:
`this.props.value` did not work as the props does not seem to have value, but I could use label as `formData`. Let me know if the update addresses your comment.